Build a regression model.

In [1]:
# import
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm

In [2]:
data = pd.read_csv('final_df.csv')

In [21]:
data.head()

station_id  Latitude   Longitude  Num_of_Slots  \
0  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
1  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
2  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
3  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
4  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   

                     fsq_id    fsq_lat    fsq_long       id           name  \
0  51b201d27dd249ae714ba728  49.263877 -123.105307  13003.0            Bar   
1  4c0b1f677e3fc928cdebf482  49.295623 -123.150768  16003.0          Beach   
2  50d4bd11e4b0b55bb4b657a3  49.266477 -123.140728  13002.0         Bakery   
3  4b77ba28f964a52064a82ee3  49.279619 -123.101293  13009.0   Cocktail Bar   
4  4bf046d53a002d7fb4eb85a4  49.295495 -123.150919  18075.0  Swimming Pool   

   popularity  rating                                            address  
0    0.945327     9.3  15 8th Ave W (btwn Manitoba & Ontario), Vancou...  
1    0.989325     9.1  8501 Stanley Park Dr (at N Lagoon Dr), Vancouv...  
2    0.965011     9.0   2150 Fir St (at W 6th Ave), Vancouver BC V6J 3B5  
3    0.971802     9.0  135 Keefer St (btwn Columbia & Main), Vancouve...  
4    0.927823     8.9  8501 Stanley Park Dr (at N Lagoon Dr), Vancouv...

In [24]:
# Drop rows with missing values in x and y variables
data_cleaned = data.dropna(subset=['rating', 'id'])
data_cleaned.head()

station_id  Latitude   Longitude  Num_of_Slots  \
0  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
1  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
2  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
3  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   
4  13021bba7bbeca1ccd8726ca17da59e6  49.25094 -123.101306            14   

                     fsq_id    fsq_lat    fsq_long       id           name  \
0  51b201d27dd249ae714ba728  49.263877 -123.105307  13003.0            Bar   
1  4c0b1f677e3fc928cdebf482  49.295623 -123.150768  16003.0          Beach   
2  50d4bd11e4b0b55bb4b657a3  49.266477 -123.140728  13002.0         Bakery   
3  4b77ba28f964a52064a82ee3  49.279619 -123.101293  13009.0   Cocktail Bar   
4  4bf046d53a002d7fb4eb85a4  49.295495 -123.150919  18075.0  Swimming Pool   

   popularity  rating                                            address  
0    0.945327     9.3  15 8th Ave W (btwn Manitoba & Ontario), Vancou...  
1    0.989325     9.1  8501 Stanley Park Dr (at N Lagoon Dr), Vancouv...  
2    0.965011     9.0   2150 Fir St (at W 6th Ave), Vancouver BC V6J 3B5  
3    0.971802     9.0  135 Keefer St (btwn Columbia & Main), Vancouve...  
4    0.927823     8.9  8501 Stanley Park Dr (at N Lagoon Dr), Vancouv...

In [30]:
# Set independent variable, x
x = data_cleaned['Num_of_Slots']

# Set dependent variable, y
y = data_cleaned['rating']

# add a constant
x = sm.add_constant(x) 

# Set linear regression model
lin_reg = sm.OLS(y, x)


In [31]:
model = lin_reg.fit()
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                 rating   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                       nan
Date:                Sun, 02 Jul 2023   Prob (F-statistic):                nan
Time:                        22:31:06   Log-Likelihood:                -244.40
No. Observations:                 182   AIC:                             490.8
Df Residuals:                     181   BIC:                             494.0
Df Model:                           0                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Num_of_Slots     0.5297      0.005    107.648   

Provide model output and an interpretation of the results. 

In [32]:
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                 rating   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                       nan
Date:                Sun, 02 Jul 2023   Prob (F-statistic):                nan
Time:                        22:31:06   Log-Likelihood:                -244.40
No. Observations:                 182   AIC:                             490.8
Df Residuals:                     181   BIC:                             494.0
Df Model:                           0                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Num_of_Slots     0.5297      0.005    107.648   

The results of the OLS regression analysis indicate that only the variable Num_of_Slots is considered in the model. The R-squared value of 0.000 suggests that the independent variable(s) do not explain any variation in the rating. The adjusted R-squared value is also 0.000. The F-statistic, which tests the overall significance of the regression model, is undefined in this case.

The coefficient table shows that for each unit increase in Num_of_Slots, we expect a 0.5297 change in the rating. However, specific information such as the standard error, confidence intervals, and p-values for Num_of_Slots is not provided in the output. This limits our ability to find meaningful conclusions from the regression analysis.

# Stretch

How can you turn the regression model into a classification model?

To convert a regression model into a classification model, set cutoff points or categories for predicted values. Choose the best method for your data and problem. Evaluate performance using metrics like accuracy, precision, recall, and F1-score. Adjust the approach based on specific needs.